src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -552,17 +552,14 @@ internal void ValidateCanBeUsedForPropertyMetadataSerialization()
             {
                 VerifyMutable();
 
-                if (value is not null)
+                if (Kind != JsonTypeInfoKind.Object)
```

Comment:
Similarly, this is a change to make the property setter more consistent with the other enum policy setters.